### ISSUE TITLE
[PROVIDER][APPROLE] Error: "auth_login" are not expected here 

### ISSUE THREAD

**Issue by ichasco** [terraform-providers/terraform-provider-vault#879]
### Terraform Version

Terraform v0.13.3
+ provider registry.terraform.io/hashicorp/aws v3.11.0
+ provider registry.terraform.io/hashicorp/local v2.0.0
+ provider registry.terraform.io/hashicorp/template v2.2.0

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```
provider "vault" {
  version = "~> 2.0.0"
  alias = "approle"
  address = var.vault_url
  
  auth_login {
    path = "auth/approle/login"
    parameters = {
      role_id   = var.vault_role_id
      secret_id = var.vault_secret_id
    }
  }
}
```
### Expected Behavior
Login with approle

### Actual Behavior
Error
```
Error: Unsupported block type

  on providers.tf line 12, in provider "vault":
  12:   auth_login {

Blocks of type "auth_login" are not expected here.
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

**Comment by catsby** [comment]
Hello! I'm sorry to say that the `auth_login` block was not added to the Vault Provider until [version 2.4.0][v240] (specifically in https://github.com/terraform-providers/terraform-provider-vault/pull/552). In order to use `auth_login`, please upgrade your version of the Vault provider from `2.0.0` you have shown above. 

[v240]: https://github.com/terraform-providers/terraform-provider-vault/blob/24ffa902d8e1d697c435aec24e415148f84d6db4/CHANGELOG.md#240-october-11-2019

**Comment by ichasco** [comment]
Is there any way to update de provider? I have tried like the documentations says and I got the next error:
```
Error: Failed to query available provider packages

Could not retrieve the list of available versions for provider
hashicorp/vault: no available releases match the given constraints 2.14.0, ~>
2.0.0, ~> 2.0.0, ~> 2.0.0, ~> 2.0.0
```
My code:
```
terraform {
  required_providers {
    vault = {
      source = "hashicorp/vault"
      version = "2.14.0"
    }
  }
}

provider "vault" {
  alias   = "token"
}

provider "vault" {
  alias = "approle"
  address = var.vault_url
  
  auth_login {
    path = "auth/approle/login"
    parameters = {
      role_id   = var.vault_role_id
      secret_id = var.vault_secret_id
    }
  }
}
```
